src/main/java/com/bootcamp/demo/service/BookingService.java

```diff
@@ -0,0 +1,166 @@
+     */
+    @Override
+    public LinkedHashSet<Booking> getAllBookings() throws ExecutionException, InterruptedException {
+        LinkedHashSet<Booking> allBookings;
+        return allBookings = db.collection(COLLECTION_PATH)
+                .get()
+                .get()
+                .getDocuments()
+                .stream().map(d -> d.toObject(Booking.class))
+                .collect(Collectors.toSet())
+                .stream().sorted(Comparator.comparing(Booking::getStartDate))
+                .collect(Collectors.toCollection(LinkedHashSet::new));
+    }
+
+    /**
+     * Creates a new document containing the given Booking type param.
+     */
+    @Override
+    public String createBooking(final Booking booking) throws ExecutionException, InterruptedException, IllegalArgumentException {
+        if (booking == null) throw new IllegalArgumentException("Parameter 'booking' cannot be null");
+        return db.collection(COLLECTION_PATH)
+                .document(booking.getId())
+                .set(booking)
+                .get()
+                .getUpdateTime()
+                .toString();
+    }
+
+    @Override
+    public String deleteBooking(final String bookingId) throws ExecutionException, InterruptedException {
+        if (bookingId == null) throw new IllegalArgumentException("Parameter 'bookingId' cannot be null");
+        return db.collection(COLLECTION_PATH)
+                .document(bookingId)
+                .delete()
+                .get()
+                .getUpdateTime()
+                .toString();
+    }
+
+    @Override
+    public String updateBooking(String bookingId, String endDate, PaymentStatus newStatus) throws ExecutionException, InterruptedException {
+        if (bookingId == null) throw new IllegalArgumentException("Parameter 'bookingId' cannot be null");
+        return db.collection(COLLECTION_PATH)
+                .document(bookingId)
+                .update("endDate", endDate, "paymentStatus", newStatus)
+                .get()
+                .getUpdateTime()
+                .toString();
+    }
+
+    public Scooter bookAScooter(final Double lat, final Double longitude) throws ExecutionException, InterruptedException {
+        Random random = new Random();
+        Location userLocation = new Location(lat, longitude);
+        Set<Scooter> allAvailableScooters = scooterService.getAvailableScooters(userLocation, 10.0);
+        if (allAvailableScooters.size() > 0) {
+            int randomIndexScooter = random.nextInt(allAvailableScooters.size());
+            Scooter[] listOfScooters = allAvailableScooters.toArray(new Scooter[allAvailableScooters.size()]);
+            Scooter pickedScooter = listOfScooters[randomIndexScooter];
+            pickedScooter.setStatus(ScooterStatus.RESERVED);
+            scooterService.updateScooter(pickedScooter.getSerialNumber(), pickedScooter.getCurrentLocation(), pickedScooter.getStatus(), pickedScooter.getBattery().getLevel());

```

Comment:
When booking a scooter isn't updating status sufficient?
Besides, currentLocation and battery are "empty" updates in the above code.